onnx/defs/nn/defs.cc

```diff
@@ -2487,4 +2487,194 @@ ONNX_OPERATOR_SET_SCHEMA(
         )ONNX"
         ));
 
+void col2imShapeInference(InferenceContext& ctx) {
+  propagateElemTypeFromInputToOutput(ctx, 0, 0);
+  // we need at least three inputs to have a shape for this inference.
+  if (!hasNInputShapes(ctx, 3)) {
+    return;
+  }
+
+  // validating 'input' input
+  auto input_shape = ctx.getInputType(0)->tensor_type().shape();
+  if ((input_shape.dim_size() < 2) || (input_shape.dim_size() > 3)) {
+    fail_shape_inference("input tensor contains partial shape. It must contain either 2 or 3 elements");
+  }
+
+  // validating 'output_shape' input
+  std::vector<int64_t> output_shape;
+  const TensorProto* temp_output_shape = ctx.getInputData(1);
+  if (temp_output_shape) {
+    auto target_shape = ParseData<int64_t>(temp_output_shape);
+    for (auto val : target_shape) {
+      output_shape.push_back(val);
+    }
+    if (output_shape.size() != 2) {
+      fail_shape_inference("output_shape tensor contains partial shape for output. It must contain either 2 or 3 elements");
+    }
+  } else {
+    // Only 2D - 3D output tensors are supported
+    if (ctx.getInputType(1)->tensor_type().shape().dim(0).dim_value() != 2) {
```

Comment:
same as above.

Reply:
You are assuming the shapes are known... this will fail if exact value is not known instead a symbol is present in dim_param

This is what I would do:
```
// validating 'output_shape' input
if (ctx.getInputType(1)->tensor_type().shape().dim_size() != 1) {
fail_shape_inference("output_shape should be 1D");
}

std::vector<int64_t> output_shape;
const TensorProto* temp_output_shape = ctx.getInputData(1);
if (temp_output_shape) {
  output_shape = ParseData<int64_t>(temp_output_shape);
  if (output_shape.size() != 2) {
      fail_shape_inference("output_shape tensor contains partial shape for output. It must contain 2 elements");
  }
```